mars/dataframe/base/get_dummies.py

```diff
@@ -0,0 +1,283 @@
+        else:
+            if op.columns:
+                encoding_columns = op.columns
+            else:
+                encoding_columns = []
+                for idx, dtype in enumerate(inp.dtypes.values):
+                    if dtype.kind in _encoding_dtype_kind:
+                        column = inp.dtypes.index[idx]
+                        encoding_columns.append(column)
+            # reindex, make encoding columns in the end of dataframe, to keep pace with pandas.get_dummies
+            total_columns = list(inp.columns.to_pandas().array)
+            for col in encoding_columns:
+                total_columns.remove(col)
+            total_columns.extend(encoding_columns)
+            inp = yield from recursive_tile(inp[total_columns])
+
+            chunks = []
+            prefix = op.prefix
+            column_to_prefix = dict()
+            for c in inp.chunks:
+                chunk_op = op.copy().reset_key()
+                chunk_op.columns = []
+                if isinstance(chunk_op.prefix, list):
+                    chunk_op.prefix = []
+                chunk_param = c.params
+                chunk_param['shape'] = (np.nan, np.nan)
+                chunk_columns = c.dtypes.index
+                c = [c]
+                for chunk_column in chunk_columns:
+                    if chunk_column in encoding_columns:
+                        chunk_op.columns.append(chunk_column)
+                        unique_chunk = yield from recursive_tile(unique(inp[chunk_column]))
+                        c.extend(unique_chunk.chunks)
+                        if isinstance(prefix, list):
+                            if chunk_column in column_to_prefix.keys():
+                                chunk_op.prefix.append(column_to_prefix[chunk_column])
+                            else:
+                                column_to_prefix[chunk_column] = prefix[0]
+                                chunk_op.prefix.append(prefix[0])
+                                prefix = prefix[1:]
+                chunk = chunk_op.new_chunk(c, kws=[chunk_param])
+                chunks.append(chunk)
+
+            new_op = op.copy()
+            kw = out.params.copy()
+            kw['chunks'] = chunks
+            kw['nsplits'] = (tuple([np.nan] * inp.chunk_shape[0]), tuple([np.nan] * inp.chunk_shape[1]))
+            return new_op.new_dataframe(op.inputs, kws=[kw])
+
+    @classmethod
+    def execute(cls, ctx, op):
+        inp = ctx[op.inputs[0].key]
+        result_length = inp.shape[0]
+        unique_inputs = []
+        for unique_input in op.inputs[1:]:
+            unique_inputs.append(ctx[unique_input.key])
+
+        if unique_inputs:
+            # make all unique_input's length the same, then get a dataframe
+            max_length = len(max(unique_inputs, key=len))
+            for unique_input in unique_inputs:
+                while len(unique_input) < max_length:
+                    unique_input.append(unique_input[0])
```

Comment:
unique_inputs = [unique_list + [unique_list[0]] * (max_length - len(unique_list)) for unique_list in unique_inputs]